fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNavigationMiddleware.kt

```diff
@@ -42,10 +52,50 @@ class ReviewQualityCheckNavigationMiddleware(
         when (action) {
             is ReviewQualityCheckAction.OpenLink -> {
                 when (action.link) {
-                    is ReviewQualityCheckState.LinkType.ExternalLink -> openLink(action.link.url, true)
-                    is ReviewQualityCheckState.LinkType.AnalyzeLink -> openLink(action.link.url, false)
+                    is LinkType.ExternalLink -> {
+                        selectOrAddUseCase.invoke(url = action.link.url)
+                    }
+
+                    is LinkType.AnalyzeLink -> {
+                        loadUrlUseCase.invoke(url = action.link.url)
+                    }
                 }
+                navController.navigate(NavGraphDirections.actionGlobalBrowser(null))
             }
         }
     }
+
+    /**
+     * Types of links that can be opened from the review quality check feature.
+     */
+    sealed class LinkType {
```

Comment:
I think we can simplify this by using string constants for urls. If we need to use `context` for sumo links, we can map it in the `processAction` function.